plutus-core/plutus-core/src/PlutusCore/Constant/Kinded.hs

```diff
@@ -23,7 +25,11 @@ class KnownKind k where
     knownKind :: SingKind k
 
 -- | Plutus only supports lifted types, hence the equality constraint.
+#if MIN_VERSION_ghc_prim(0,8,0)
+instance rep ~ LiftedRep => KnownKind (TYPE rep) where
```

Comment:
This compiles fine on GHC 8 for me with `DataKinds`, I don't think you need the CPP.

Reply:
It gave me a warning IIRC.

Reply:
I think turning off the warning is fine, it's a dumb warning.

Reply:
They're even going to turn it off by default in newer GHCs, I think.